governance/handler.go

```diff
@@ -341,10 +339,8 @@ func (gov *Governance) removePreviousVote(valset istanbul.ValidatorSet, votes []
 			ret = append(votes[:idx], votes[idx+1:]...)
 			if gov.isGovernanceModeSingleOrNone(governanceMode, governingNode, gVote.Validator) ||
 				(governanceMode == params.GovernanceMode_Ballot && currentVotes <= valset.TotalVotingPower()/2) {
-				if v, ok := gov.changeSet[vote.Key]; ok && v == vote.Value {
-					gov.mu.Lock()
-					delete(gov.changeSet, vote.Key)
-					gov.mu.Unlock()
+				if v, ok := gov.changeSet.GetValue(GovernanceKeyMap[vote.Key]); ok && v == vote.Value {
```

Comment:
If there is not a global lock, there might be a race between GetValue() and RemoveItem().

For example,
1) goroutine A: GetValue() is ok and v == vote.Value
2) goroutine B: Updated by SetValue()
3) goroutine A: RemoveItem() --> wrong deletion ( v != vote.Value)

Reply:
@jeonghonah I think this is a good point and it is revealing another issue. Let's have some discussion f2f.